### ISSUE TITLE
inserts newlines in settings plugins section on windows

### ISSUE THREAD

**Issue by NikkyAI** [jmfayard/refreshVersions#375]
- [x] I have read the [guidelines for submitting issues](https://jmfayard.github.io/refreshVersions/contributing/submitting-issues/)
## 🐛 Describe the bug

refreshVersion inserts system depednent newline before the available versions for settings plugins
on windows the function `removeCommentsAddedByUs()` does not correctly strip newlines, leaving a extra `\r` each time the task runs, leading to extra newlines

```
plugins {
    id("com.gradle.enterprise") version "3.6.1"


////                        # available:"3.6.2"
////                        # available:"3.6.3"
    id("de.fayard.refreshVersions") version "0.10.1"
}
```

possible options are:
- not using system dependent newline here: https://github.com/jmfayard/refreshVersions/blob/main/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/SettingsPluginsUpdater.kt#L131 instead use `append("\n")`
- using the `System.lineSeparator()` everywhere (but also handle if the file is LF on windows)

this patch seems to fix it for me

```diff
diff --git a/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/SettingsPluginsUpdater.kt b/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/SettingsPluginsUpdater.kt
index 281e4896..4d95b64e 100644
--- a/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/SettingsPluginsUpdater.kt
+++ b/plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/SettingsPluginsUpdater.kt
@@ -128,7 +128,7 @@ internal object SettingsPluginsUpdater {
             val availableVersionsComments = buildString {
                 val padding = offsetOfCurrentVersionFromLineStart - 8
                 pluginWithVersionCandidates.versionsCandidates.forEach { versionCandidate ->
-                    appendln()
+                    append("\n")
                     append("//// ")
                     append(availableComment.padStart((padding).coerceAtLeast(0)))
                     append(':')

```

i doubt anyone would need to have the file with windows lineendings, git ahould already do that conversion anyways

## 📱 Tech info
- OS: Windows 10
- IDE: IntelliJ IDEA 2021.1.2 (Ultimate Edition)

**Comment by NikkyAI** [comment]
for debugging this i added 
```

            System.err.println("currentVersionEndIndex: $currentVersionEndIndex")
            System.err.println("indexOfNextNewLine: $indexOfNextNewLine")
            val inputEscaped = this.toString()
                .replace("\r", "\\r")
                .replace("\n", "\\n\n")
            System.err.println("inputEscaped: '$inputEscaped'")
            System.err.println("adding comments at $indexOfNextNewLine: '$availableVersionsComments'")
            System.err.println("currentVersionPadding: $currentVersionPadding")
```
to the end of forEach in `insertAvailableVersionComments`

hopefully that helps a bit

when the file is switched to CLRF
this prints:

```
plugins {\r\n
    id("com.gradle.enterprise") version "3.6.1"\r\n
////                        # available:"3.6.2"\n
////                        # available:"3.6.3"\n
    id("de.fayard.refreshVersions")\r\n
}\r\n
```
and it seems to continue working as well

**Comment by LouisCAD** [comment]
Hello, does this still reproduce with 0.30.2?

**Comment by NikkyAI** [comment]
it seems fixed with 0.30.2

**Comment by LouisCAD** [comment]
Great! Thanks for confirming what I assumed.
